packages/type-safe-api/src/construct/integrations/error-integration-response.ts

```diff
@@ -0,0 +1,95 @@
+/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0 */
+import {
+  ApiGatewayIntegrationResponse,
+  IntegrationRenderProps,
+} from "./integration";
+import { generateCorsResponseParameters } from "../prepare-spec-event-handler/prepare-spec";
+
+/**
+ * An error integration response. You can extend this to implement your own error integration response
+ */
+export abstract class ErrorIntegrationResponse {
+  /**
+   * Render the error integration response into the Integration
+   * @param props the integration render props (same used in the Integration)
+   */
+  public abstract render(props: IntegrationRenderProps): {
+    [responseStatusPattern: string]: ApiGatewayIntegrationResponse;
+  };
+}
+
+/**
+ * An error integration response that returns no error responses
+ */
+export class NoneErrorIntegrationResponse extends ErrorIntegrationResponse {
+  public render(): {
```

Comment:
Looks like the JSII build failed with the error:
```
error JSII5005: "@aws/type-safe-api.NoneErrorIntegrationResponse#render" has 0 parameters when overriding @aws/type-safe-api.ErrorIntegrationResponse. It should accept 1 parameters
```

I think you might just need to do:

```ts
public render(_props: IntegrationRenderProps): {
```

😄

Reply:
You're right I didn't understand where this error came from, thanks for the fix!

Now, tests run but I got the following error:

```
@aws/type-safe-api:   ● Test suite failed to run
@aws/type-safe-api:     Cannot find module 'aws-cdk-lib/aws-s3/lib/perms' from 'src/construct/integrations/s3.ts'
@aws/type-safe-api:     Require stack:
@aws/type-safe-api:       src/construct/integrations/s3.ts
@aws/type-safe-api:       src/construct/integrations/integrations.ts
@aws/type-safe-api:       src/construct/integrations/index.ts
@aws/type-safe-api:       src/construct/index.ts
@aws/type-safe-api:       src/index.ts
@aws/type-safe-api:       test/project/model/type-safe-api-model-project.test.ts
@aws/type-safe-api:       4 | import { Grant, IRole, Role, ServicePrincipal } from "aws-cdk-lib/aws-iam";
@aws/type-safe-api:       5 | import { IBucket } from "aws-cdk-lib/aws-s3";
@aws/type-safe-api:     > 6 | import * as perms from "aws-cdk-lib/aws-s3/lib/perms";
@aws/type-safe-api:         | ^
@aws/type-safe-api:       7 | import * as cxapi from "aws-cdk-lib/cx-api";
@aws/type-safe-api:       8 | import { IConstruct } from "constructs";
@aws/type-safe-api:       9 | import { ErrorIntegrationResponse } from "./error-integration-response";
@aws/type-safe-api:       at Resolver._throwModNotFoundError (../../node_modules/.pnpm/jest-resolve@29.6.4/node_modules/jest-resolve/build/resolver.js:427:11)
@aws/type-safe-api:       at Object.require (src/construct/integrations/s3.ts:6:1)
@aws/type-safe-api:       at Object.require (src/construct/integrations/integrations.ts:6:1)
@aws/type-safe-api:       at Object.require (src/construct/integrations/index.ts:4:1)
@aws/type-safe-api:       at Object.require (src/construct/index.ts:6:1)
@aws/type-safe-api:       at Object.require (src/index.ts:9:1)
@aws/type-safe-api:       at Object.<anonymous> (test/project/model/type-safe-api-model-project.test.ts:4:1)
```

Maybe we're not supposed to used perms directly?

Reply:
Ahh sad times! Yeah sounds like we can't use them directly :(

Perhaps we go with the simpler `grantRead` and `grantReadWrite` option? Alternatively we could copy/paste the perms but that might get a bit annoying to maintain!